README.md

```diff
@@ -228,3 +228,125 @@ This endpoint returns JSON. An example follows below.
+
+Example payload:
+
+```
+{
+  type: "verification_approved",
+  data: {
+    user_id: "14ec6af0-12f8-4bce-a6ab-01ce87fa1812",
+  }
+}
+```
+
+### Securing webhooks
+
+#### Callback URL
+
+Fractal ID accepts the following callback URLs:
+* Secure sites (HTTPS) - for production usage.
+* Localhost URLs - for development / testing. For example: `http://localhost:3000/callback`
+
+#### Setting your secret token
+
+After Fractal provides secret token to the partner, it should set up an environment variable on its server that stores this token. Typically, this is as simple as running:
+
+```
+export SECRET_TOKEN=your_token
+```
+
+**Never** hardcode the token into your app!
+
+#### Validating payloads from Fractal ID
+
+When your secret token is set, Fractal ID uses it to create a hash signature with each payload. The hash signature is passed along with each request in the headers as `X-Fractal-Signature`. Fractal ID uses HMAC hexdigest to compute the hash.
+
+Partner should verify the signature on their server. Example implementation in Ruby:
+
+```
+
+def verify_signature
+  payload_body = request.body.read
+  signature = "sha1=" + OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new("sha1"), ENV["SECRET_TOKEN"], payload_body)
+
+  return if Rack::Utils.secure_compare(signature, request.headers["X-Fractal-Signature"])
+
+  render json: { error: "signature_mismatch" }, status: 500
+end
+```
+
+Obviously, your language and server implementations may differ than this code. There are a couple of very important things to point out, however:
+
+- No matter which implementation you use, the hash signature starts with `sha1=`, using the key of your secret token and your payload body.
+- Using a plain `==` operator is **not advised**. A method like `secure_compare` performs a "constant time" string comparison, which renders it safe from certain timing attacks against regular equality operators.
+
+### Delivery
+
+#### Content type
+
+Fractal ID uses `application/json` content type to deliver the JSON payload directly as the body of the `POST` request.
+
+#### Retrying sending notifications
+
+Fractal ID will retry to send webhooks notification if:
+
+- Client server failed to respond in 10 seconds.
+- Client server is unavailable (network errors).
+- Client server returns other response code than 200.
+
+Fractal ID uses exponential backoff to retry events.
+
+**Example retry times table:**
+
+| Retries | Next retry in seconds |
+|---------|-----------------------|
+|    1    |           10          |
+|    2    |           20          |
+|    3    |           50          |
+|    5    |          260          |
+|    10   |          1010         |
+|    50   |          25010        |
+|    100  |          86400        |
+|    200  |          86400        |
+
+#### Example delivery
```

Comment:
Maybe we should provide the secret used in this case, so that they can check their MAC checking algorithm against a known sample.

Reply:
Sure!